components/optimizer/__init__.py

```diff
@@ -0,0 +1,22 @@
+from .greedymax import GreedyMax
+from common.minimodel import Plans
+
+
+class Optimizer:
+    """
+    Entrypoint to interact with an BaseOptimizer object.
+    All algorithms need to follow the same structure to create a Plan
+    """
+
+    def __init__(self, selection, algorithm=GreedyMax(some_parameter=1)):
```

Comment:
We should talk about what the `selection` is.
I think I see below that it is a:
```
Dict[GroupID, Tuple[Group, GroupInfo]]
```
?

That seems good to me, but I am assuming that you will also need more information, such as `TargetInfo`, no?